tests/test_rrt_star_reeds_shepp.py

```diff
@@ -6,6 +6,10 @@ def test1():
     m.show_animation = False
     m.main(max_iter=5)
 
+def test2():
+    m.show_animation = False
+    m.main(max_iter=30, random_seed=8)
```

Comment:
I think this test should check the all path pitch distances should be less than a certain value to check the problematic behavior was solved.

Reply:
True, I updated the tests in test_rrt_star_reeds_shepp.py in new commits, this test2() is removed and two tests with distance checks were added.

It should be noted that, I believe it is the error caused by the precision of the python calculation that causes the distance between two points to exceed the step size in some cases.

Here is the situation under step_size=0.2:

<img width="298" alt="Snipaste_2023-04-02_23-34-12" src="https://user-images.githubusercontent.com/39330401/229405732-a7745539-e1d1-4220-8b51-bb62ba2186c2.png">

Similar cases are found for step_size = 20 and step_size = 0.02, and the error is always less than 1e-14, independent of the size of step_size.

So as I commented in the code,  step + 0.00000000000001 is set as the upper limit of distance between every two points along the path, which allows these acceptable errors to be ignored.